### ISSUE TITLE
Module 2: UNet Architecture - Encoder-Decoder (2 Pomodoros)

### ISSUE THREAD

**Issue by gregory1506** [gregory1506/arc-pytorch-prep#2]
## Module 2: UNet Architecture - Encoder-Decoder Design
**Duration**: 2 Pomodoros (50 minutes)  
**Status**: 🔄 Not Started  
**Priority**: High
**Depends on**: #1

### Learning Objectives
By the end of this module, students will be able to:
1. Explain UNet architecture components
2. Implement encoder (contracting path) with convolutions and pooling
3. Implement decoder (expanding path) with upsampling
4. Add skip connections to preserve spatial information
5. Calculate output dimensions given input size

### Pomodoro 1: Teaching + Architecture Design (25 min)
- [ ] **Teaching (10 min)**: UNet architecture, skip connections, semantic segmentation
- [ ] **Review (5 min)**: Look at src/models/unet.py skeleton
- [ ] **Architecture Planning (5 min)**: Discuss channel dimensions, depth
- [ ] **Discussion (5 min)**: Why UNet for remote sensing?

### Pomodoro 2: Implementation + Testing (25 min)
- [ ] **Code (15 min)**: Fill in unet.py sections
  - TODO 2.1: Implement DoubleConv block
  - TODO 2.2: Build encoder with max pooling
  - TODO 2.3: Implement decoder with upsampling
  - TODO 2.4: Add skip connection concatenation
  - TODO 2.5: Final 1x1 convolution for class predictions
- [ ] **Test (5 min)**: Run pytest tests/test_module_02_unet.py -v
- [ ] **Shape Verification (5 min)**: Confirm output matches input spatial dims

### Files to Edit
- src/models/unet.py

### Test Command
pytest tests/test_module_02_unet.py -v

### Teaching Notes
See TEACHING.md#module-2-unet-architecture for:
- Encoder-decoder architecture diagrams
- Skip connection visualizations
- Common pitfalls (concatenation dimension, spatial sizes)

### Acceptance Criteria
- [ ] DoubleConv block maintains spatial dimensions
- [ ] Encoder reduces spatial dims and increases channels correctly
- [ ] Decoder upsamples and concatenates skip connections
- [ ] Output shape matches input spatial dimensions
- [ ] Output has no activation (returns logits)
- [ ] All tests in test_module_02_unet.py pass

### Resources
- U-Net Paper: https://arxiv.org/abs/1505.04597
- Understanding U-Net: https://towardsdatascience.com/understanding-u-net-6127b961c483

---
**Labels**: tutorial, module-2, teaching, coding, testing
